### PR TITLE
Try all possible addresses to bind and connect

### PR DESCRIPTION
**Pull request by anisimkov**
S304-032

### CODE REVIEW

File: config/src/aws-net-std__gnat.adb [hunk 1]
```diff
@@ -114,51 +108,64 @@ package body AWS.Net.Std is
...
       if Socket.S /= null then
          Socket := Socket_Type'(Net.Socket_Type with others => <>);
       end if;
 
       Socket.S := new Socket_Hidden;
 
-      Addr_Info := Get_Inet_Addr (Host, Family, Passive => True);
+      declare
+         use GNAT.Sockets;
+         Error  : Exceptions.Exception_Occurrence;
```
Comment (TurboGit) [inline config/src/aws-net-std__gnat.adb]: Align on ':'

File: config/src/aws-net-std__gnat.adb [hunk 2]
```diff
@@ -218,51 +224,80 @@ package body AWS.Net.Std is
...
+                  Connect_Socket (Socket.S.FD, Sock_Addr);
+               exception
+                  when E : GNAT.Sockets.Socket_Error =>
+                     --  Ignore EWOULDBLOCK and EINPROGRESS errors, because we
+                     --  are using none blocking connect.
+                     --  ??? Note, we should change this when GNAT will support
+                     --  Non-blocking connect.
+
+                     case Resolve_Exception (E) is
+                     when Operation_Now_In_Progress
```
Comment (TurboGit) [inline config/src/aws-net-std__gnat.adb]: when should be indented.

File: config/src/aws-net-std__gnat.adb [hunk 3]
```diff
@@ -218,51 +224,80 @@ package body AWS.Net.Std is
...
-
-            case Resolve_Exception (E) is
-               when Operation_Now_In_Progress
-                    | Resource_Temporarily_Unavailable => null;
-               when others =>
-                  Raise_Error (Exception_Message (E));
-            end case;
-      end;
+         for Addr_Info of Addresses loop
+            begin
```
Comment (TurboGit) [inline config/src/aws-net-std__gnat.adb]: this block is big, maybe we should name it?
Check_Address : begin
 ...
end Check_Address;

File: config/src/aws-net-std__gnat.adb [hunk 4]
```diff
@@ -218,51 +224,80 @@ package body AWS.Net.Std is
...
+                  if Close_On_Exception then
+                     Sockets.Close_Socket (Socket.S.FD);
+                     Close_On_Exception := False;
+                  end if;
+                  Save_Occurrence (Keep_Excp, E);
+            end;
+         end loop;
+
+         Raise_Error (Exception_Message (Keep_Excp));
+      end;
```
Comment (TurboGit) [inline config/src/aws-net-std__gnat.adb]: end Check_Address;

File: config/src/aws-net-std__gnat.adb (hunk 3 above)
Comment (anisimkov) [reply config/src/aws-net-std__gnat.adb]: Done

**Comment by anisimkov** [comment]
All suggestions applied

File: config/src/aws-net-std__gnat.adb [hunk 5]
```diff
@@ -114,51 +108,64 @@ package body AWS.Net.Std is
...
       if Socket.S /= null then
          Socket := Socket_Type'(Net.Socket_Type with others => <>);
       end if;
 
       Socket.S := new Socket_Hidden;
 
-      Addr_Info := Get_Inet_Addr (Host, Family, Passive => True);
+      declare
+         use GNAT.Sockets;
+         Error : Exceptions.Exception_Occurrence;
```
Comment (TurboGit) [inline config/src/aws-net-std__gnat.adb]: This one also needs alignment.

Comment (anisimkov) [reply config/src/aws-net-std__gnat.adb]: Done, renamed Error to Keep_Excp to be consistent with another similar block.

**Pull request merged**